### PR TITLE
Adds new Donk-Pockets!

### PR DESCRIPTION
**Pull request by Coconutwarrior97**
## About The Pull Request

After a generous donation from the S%n#@!&e, Donk Co. is pleased to announce its new line of products!

Thanks to Farquaar for the sprites and some of the coding work!

## Why It's Good For The Game
Variety is the spice of life, and the new donk pockets are here to help!
## Changelog
:cl:
add: Added several new donk pocket types, and boxes, including sprites for each of course. Spicy-Pocket,Teriyaki-Pocket, Pizza-Pocket,Honk-Pocket,Gondola-Pocket and Berry-Pocket. This includes re-organizing them from the OTHER section in snacks_pastry.dm to a donk pocket section. Thanks to Farquaar for the box icons and some of the code! As well as latots for the donk pocket icons.
add: Added a Donk Pocket Variety crate, for 2000 credits, which includes 3 randomly selected donk pocket box types, excluding Gondola-Pocket.
add: Added a recipe for each new Donk Pocket type.
add: Added the new Donk Pocket types to the maint loot table.
add: Donk Pocket loot spawner object, credit to TheVekter for that, which was then added into the kitchens of all the main maps.
tweak: All Donk Pockets now follow the same naming scheme of X-Pocket.

/:cl:

image

### CODE REVIEW

**Comment by Farquaar** [comment]
At last, our dreams shall become a reality, Coconut Combatant.

**Comment by TheVekter** [comment]
Needs to have a random table set up to determine which one spawns in the kitchen.

**Comment by Coconutwarrior97** [comment]
@TheVekter 

I'm pretty sure the donk pocket placement in departments is handled via mapping, yea? 
Also, the Gondola-Pocket does turn you into a Gondola. However the only way to get it is if you already have the means to turn people into Gondolas or Admin Fuckery.

**Comment by TheVekter** [comment]
> I'm pretty sure the donk pocket placement in departments is handled via mapping, yea?

Yeah, but you can create random spawning elements that pull from a weighted table. It's how come all the shit in maintenance changes every round. If you wanna hop on #coderbus I can explain it better in Discord.

File: code/_globalvars/lists/maintenance_loot.dm [hunk 1]
```diff
@@ -213,6 +213,14 @@ GLOBAL_LIST_INIT(uncommon_loot, list(//uncommon: useful items
 	list(//food
 		/obj/item/reagent_containers/food/snacks/canned/peaches/maint = 1,
 		/obj/item/storage/box/donkpockets = 1,
+		list(//Donk varities
```
Comment (TheVekter) [inline code/_globalvars/lists/maintenance_loot.dm]: Varieties

File: code/_globalvars/lists/maintenance_loot.dm [hunk 2]
```diff
@@ -213,6 +213,14 @@ GLOBAL_LIST_INIT(uncommon_loot, list(//uncommon: useful items
...
+			/obj/item/storage/box/donkpockets/donkpocketspicy = 1,
+			/obj/item/storage/box/donkpockets/donkpocketteriyaki = 1,
+			/obj/item/storage/box/donkpockets/donkpocketpizza = 1,
+			/obj/item/storage/box/donkpockets/donkpocketgondola= 1,
```
Comment (TheVekter) [inline code/_globalvars/lists/maintenance_loot.dm]: This makes it possible for gondola pockets to spawn in maint. If that's not intentional you should probably 86 it.

File: code/_globalvars/lists/maintenance_loot.dm (hunk 1 above)
Comment (Coconutwarrior97) [reply code/_globalvars/lists/maintenance_loot.dm]: Thanks, Ive fixed this.

File: code/_globalvars/lists/maintenance_loot.dm (hunk 2 above)
Comment (Coconutwarrior97) [reply code/_globalvars/lists/maintenance_loot.dm]: Good find! Its been removed.

**Comment by Akrilla** [comment]
image
:worry:

**Comment by Coconutwarrior97** [comment]
I have the correct map merger ready but python hates me and I am currently trying to fix some things so I can properly commit the changes.

**Comment by Iatots** [comment]
These sprites look rather noisy, I think it would look better if you kept the pastry band neutral and have the filling be the only difference between the sprites.

**Comment by Grazyn** [comment]
Yeah the Gondola one looks better with the face on it rather than on the wrapping since the donk itself already looks like a Gondola

File: code/modules/food_and_drinks/food/snacks_pastry.dm [hunk 3]
```diff
@@ -439,6 +426,128 @@
...
+	bonus_reagents = list(/datum/reagent/medicine/omnizine = 1, /datum/reagent/consumable/laughter = 3)
+	list_reagents = list(/datum/reagent/consumable/nutriment = 4, /datum/reagent/medicine/omnizine = 3, /datum/reagent/consumable/banana = 4, /datum/reagent/consumable/laughter = 3)
+	tastes = list("dough" = 2, "children's antibiotics" = 1)
+	foodtype = GRAIN
+
+/obj/item/reagent_containers/food/snacks/donkpocket/berry
+	name = "\improper Berry-pocket"
+	desc = "A relentlessly sweet donk-pocket first created for use in Operation Dessert Storm."
+	icon_state = "donkpocketberry"
+	list_reagents = list(/datum/reagent/consumable/nutriment = 4, /datum/reagent/consumable/berryjuice = 3)
```
Comment (ATH1909) [inline code/modules/food_and_drinks/food/snacks_pastry.dm]: Time to hunt in maint for these as a bartender; berry juice is the reagent that keeps the bartender from making most of their cool healing drinks.

**Comment by Farquaar** [comment]
@latots
I like those sprites. The Donk-Sprites that I made were some of my first attempts at spriting, so I wouldn't be terribly offended if they were scrapped.

**Comment by Shadowflame909** [comment]
I feel like I've seen this before...

Deja-Vu!

**Comment by Coconutwarrior97** [comment]
@Iatots 
I asked a couple people on the tg discord their thoughts and they seemed to agree with you, if you want to send me the .dmi for those donks I'd be happy to add them in and add you to the CL. My discord is CoconutMage#8333 .

**Comment by Krysonism** [comment]
I like farquaar's sprites better, I like the detailing on the wrapper.

**Comment by Coconutwarrior97** [comment]
@Krysonism 
The wrapper detail in general felt a bit too noisy imo, a bit blurry at times and too much going on, especially the berry and teriyaki pocket. I prefer this since it just looks cleaner to me.

**Comment by Farquaar** [comment]
@Kyrsonism
I appreciate the compliment man, but the wrapper detailing doesn't really make much sense in hindsight. After all, microwavable snack foods typically have plain wrappers anyway, and having each Donk use the same wrapper makes them more easily distinguishable as the same class of food.

**Comment by delaron** [comment]
Syringe Inject some Lime and Vodka to the Pizza Donk Pocket and you will have a good time!

**Comment by Iatots** [comment]
I thought it was a dough band like pigs in a blanket...

**Comment by Farquaar** [comment]
Donk-pockets are loosely based off of hot pockets, which are brown and have a little wrapper for easy microwaving. (https://grist.files.wordpress.com/2014/02/hot-pockets-flickr-nestle.jpg?w=1024)

**Comment by cacogen** [comment]
Those are truly terrible and that cardboard thing does nothing to prevent you burning your fingertips. I always imagined donk pockets to be sweet like twinkies for some reason despite understanding the reference

File: code/game/objects/items/storage/boxes.dm [hunk 4]
```diff
@@ -438,6 +438,96 @@
...
+/obj/item/storage/box/donkpockets/donkpocketspicy
+	name = "box of spicy-flavoured donk-pockets"
+	desc = "<B>Instructions:</B> <I>Heat in microwave. Product will cool if not eaten within seven minutes.</I>"
+	icon_state = "donkpocketboxspicy"
+	illustration=null
+
+/obj/item/storage/box/donkpockets/donkpocketspicy/ComponentInitialize()
+	. = ..()
+	var/datum/component/storage/STR = GetComponent(/datum/component/storage)
+	STR.set_holdable(list(/obj/item/reagent_containers/food/snacks/donkpocket))
```
Comment (Ghommie) [inline code/game/objects/items/storage/boxes.dm]: `/obj/item/storage/box/donkpockets/ComponentInitialize()` already handles the storage. Feel free to remove these lines.

File: code/game/objects/items/storage/boxes.dm [hunk 5]
```diff
@@ -438,6 +438,96 @@
...
+/obj/item/storage/box/donkpockets/donkpocketteriyaki
+	name = "box of teriyaki-flavoured donk-pockets"
+	desc = "<B>Instructions:</B> <I>Heat in microwave. Product will cool if not eaten within seven minutes.</I>"
+	icon_state = "donkpocketboxteriyaki"
+	illustration=null
+
+/obj/item/storage/box/donkpockets/donkpocketteriyaki/ComponentInitialize()
+	. = ..()
+	var/datum/component/storage/STR = GetComponent(/datum/component/storage)
+	STR.set_holdable(list(/obj/item/reagent_containers/food/snacks/donkpocket))
```
Comment (Ghommie) [inline code/game/objects/items/storage/boxes.dm]: Idem

File: code/game/objects/items/storage/boxes.dm [hunk 6]
```diff
@@ -438,6 +438,96 @@
...
+/obj/item/storage/box/donkpockets/donkpocketpizza
+	name = "box of pizza-flavoured donk-pockets"
+	desc = "<B>Instructions:</B> <I>Heat in microwave. Product will cool if not eaten within seven minutes.</I>"
+	icon_state = "donkpocketboxpizza"
+	illustration=null
+
+/obj/item/storage/box/donkpockets/donkpocketpizza/ComponentInitialize()
+	. = ..()
+	var/datum/component/storage/STR = GetComponent(/datum/component/storage)
+	STR.set_holdable(list(/obj/item/reagent_containers/food/snacks/donkpocket))
```
Comment (Ghommie) [inline code/game/objects/items/storage/boxes.dm]: same

File: code/game/objects/items/storage/boxes.dm [hunk 7]
```diff
@@ -438,6 +438,96 @@
...
+/obj/item/storage/box/donkpockets/donkpocketgondola
+	name = "box of gondola-flavoured donk-pockets"
+	desc = "<B>Instructions:</B> <I>Heat in microwave. Product will cool if not eaten within seven minutes.</I>"
+	icon_state = "donkpocketboxgondola"
+	illustration=null
+
+/obj/item/storage/box/donkpockets/donkpocketgondola/ComponentInitialize()
+	. = ..()
+	var/datum/component/storage/STR = GetComponent(/datum/component/storage)
+	STR.set_holdable(list(/obj/item/reagent_containers/food/snacks/donkpocket))
```
Comment (Ghommie) [inline code/game/objects/items/storage/boxes.dm]: ditto

File: code/game/objects/items/storage/boxes.dm [hunk 8]
```diff
@@ -438,6 +438,96 @@
...
+/obj/item/storage/box/donkpockets/donkpocketberry
+	name = "box of berry-flavoured donk-pockets"
+	desc = "<B>Instructions:</B> <I>Heat in microwave. Product will cool if not eaten within seven minutes.</I>"
+	icon_state = "donkpocketboxberry"
+	illustration=null
+
+/obj/item/storage/box/donkpockets/donkpocketberry/ComponentInitialize()
+	. = ..()
+	var/datum/component/storage/STR = GetComponent(/datum/component/storage)
+	STR.set_holdable(list(/obj/item/reagent_containers/food/snacks/donkpocket))
```
Comment (Ghommie) [inline code/game/objects/items/storage/boxes.dm]: .

File: code/game/objects/items/storage/boxes.dm [hunk 9]
```diff
@@ -438,6 +438,96 @@
...
+/obj/item/storage/box/donkpockets/donkpockethonk
+	name = "box of banana-flavoured donk-pockets"
+	desc = "<B>Instructions:</B> <I>Heat in microwave. Product will cool if not eaten within seven minutes.</I>"
+	icon_state = "donkpocketboxbanana"
+	illustration=null
+
+/obj/item/storage/box/donkpockets/donkpockethonk/ComponentInitialize()
+	. = ..()
+	var/datum/component/storage/STR = GetComponent(/datum/component/storage)
+	STR.set_holdable(list(/obj/item/reagent_containers/food/snacks/donkpocket))
```
Comment (Ghommie) [inline code/game/objects/items/storage/boxes.dm]: .

**Review by Ghommie — COMMENTED** [review]
Also those `illustration` and `desc` vars restated on each new subtype without any diference from the base type should be removed.

**Comment by kriskog** [comment]
Since the donks can be crafted, it would be nice if the boxes could too. A recipe already exists for the normal box, it should be trivial to add. See sheet_types.dm, fancy boxes.

**Comment by KomradeSpectre** [comment]
It looks like you may have ticked the _maps/*.dm files

**Comment by Coconutwarrior97** [comment]
Ive been busy and am about to be on holiday so I wont be able to work on this for a bit. Ill get back to it late this weekend. Thanks for all the help with it y'all.

File: code/game/objects/items/storage/boxes.dm (hunk 4 above)
Comment (Coconutwarrior97) [reply code/game/objects/items/storage/boxes.dm]: Done, thanks for the help!

File: code/game/objects/items/storage/boxes.dm (hunk 5 above)
Comment (Coconutwarrior97) [reply code/game/objects/items/storage/boxes.dm]: Done.

File: code/game/objects/items/storage/boxes.dm (hunk 6 above)
Comment (Coconutwarrior97) [reply code/game/objects/items/storage/boxes.dm]: Done.

File: code/game/objects/items/storage/boxes.dm (hunk 7 above)
Comment (Coconutwarrior97) [reply code/game/objects/items/storage/boxes.dm]: Done.

File: code/game/objects/items/storage/boxes.dm (hunk 8 above)
Comment (Coconutwarrior97) [reply code/game/objects/items/storage/boxes.dm]: Done.

File: code/game/objects/items/storage/boxes.dm (hunk 9 above)
Comment (Coconutwarrior97) [reply code/game/objects/items/storage/boxes.dm]: Done.

**Comment by Coconutwarrior97** [comment]
Been busy studying for exams and end of the semester stuff. My last exam is
Wednesday so I am shooting to finish this up soon after that.

**Comment by TheVekter** [comment]
@Coconutwarrior97 You mind if I take this over?

**Pull request closed without merging**